storage/oasis/runtime.go

```diff
@@ -2,95 +2,56 @@ package oasis
 
 import (
 	"context"
-	"fmt"
 
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis/nodeapi"
 	config "github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
-	connection "github.com/oasisprotocol/oasis-sdk/client-sdk/go/connection"
-	runtimeSignature "github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
-	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/evm"
 	sdkTypes "github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
 
 	"github.com/oasisprotocol/oasis-indexer/storage"
 )
 
-// RuntimeClient is a client to a ParaTime.
+// RuntimeClient is a client to a runtime. Unlike RuntimeApiLite implementations,
+// which provide a 1:1 mapping to the Oasis node's runtime RPCs, this client
+// is higher-level and provides a more convenient interface for the indexer.
+//
+// TODO: Get rid of this struct, it hardly provides any value.
 type RuntimeClient struct {
-	client  connection.RuntimeClient
-	network *config.Network
-
-	info  *sdkTypes.RuntimeInfo
-	rtCtx runtimeSignature.Context
+	nodeApi nodeapi.RuntimeApiLite
+	info    *sdkTypes.RuntimeInfo
 }
 
 // AllData returns all relevant data to the given round.
 func (rc *RuntimeClient) AllData(ctx context.Context, round uint64) (*storage.RuntimeAllData, error) {
-	blockData, err := rc.BlockData(ctx, round)
+	blockHeader, err := rc.nodeApi.GetBlockHeader(ctx, round)
 	if err != nil {
 		return nil, err
 	}
-	rawEvents, err := rc.GetEventsRaw(ctx, round)
+	rawEvents, err := rc.nodeApi.GetEventsRaw(ctx, round)
 	if err != nil {
 		return nil, err
 	}
-
-	data := storage.RuntimeAllData{
-		Round:     round,
-		BlockData: blockData,
-		RawEvents: rawEvents,
-	}
-	return &data, nil
-}
-
-// BlockData gets block data in the specified round.
-func (rc *RuntimeClient) BlockData(ctx context.Context, round uint64) (*storage.RuntimeBlockData, error) {
```

Comment:
:+1: to the former. I didn't want to do those changes now because the refactors are big enough as it is. 

I don't want to consolidate ApiLite and SourceStorage. SourceStorage used to do everything ApiLite does (RPCs), and also some data munging. The data munging is not really dependent on node version, so it didn't make much sense to create a separate cobalt and separate damask SourceStorage. Hence I abstracted the RPCs into ApiLite. Only later, when I was heavily mid-refactor, did I realize that the data munging is limited in scope and quite limited (IMO) in the value it brings, so it would've been probably smoother to first rip out that munging (either into the consumer code, or just get rid of parts of the munging and work with slightly differently-typed data downstream) and then turn SourceStorage into what ApiLite is today. C'est la vie.

But we can still get rid of the munging aspects of SourceStorage, which will then make SourceStorage completely obsolete (because munging won't be needed, and RPCs will be in ApiLite) and we can remove it. Thus effectively achieving the "ideal" setup I described above, but with a slightly painful and clumsy rename of SourceStorage into ApiLite.

I don't see these cleanups as pretty low priority though.

cc @pro-wh FYI because we've talked a little about this before.

Reply:
interesting, so we would push that munging into the analyzer. that'd be a nice refactor for later. now that we have this pretty clean download--process--db stages, we'll probably have a nice place to receive that code.

there's still this distant dream of running a class of tests on a special SourceStorage (or ApiLite) implementation that reads from disk :shrug:

Reply:
#352 is probably back to the drawing board :upside_down_face: